### PR TITLE
Unit test for Gram Matrix algorithms

### PR DESCRIPTION
**Pull request by jetSett**
- [x] test_gso
- [x] test_lll
- [x] test_bkz
- [x] test_enum
- [ ] test_cvp => postponed

Others :
- [x] fix test_gso_coherence_gram_matrix for n >= 30

### CODE REVIEW

File: tests/tools.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+from copy import copy
+
+def float_equals(x, y, epsilon=0.0001):
```
Comment (malb) [inline tests/tools.py:5]: Our testing platform kinda has this built in: https://stackoverflow.com/a/39623614/5053478

Comment (jetSett) [reply tests/tools.py]: Thank you for the insight, it is fixed in f59dfc67ef76fe3e2cfd3a2fec4c739307ef62f1

File: tests/test_gso.py [hunk 2]
```diff
@@ -109,3 +116,44 @@ def test_gso_io():
...
+                for i in range(m):
+                    for j in range(i):
+                        assert M_A.get_int_gram(i, j) == G[i, j]
+
+                # Check if computations coincide
+                for i in range(m):
+                    for j in range(i):
+                        assert (
+                            M_A.get_r(i, j) == 
+                            pytest.approx(M_G.get_r(i, j), rel=EPSILON)
```
Comment (WvanWoerden) [inline tests/test_gso.py:154]: The diagonal get_r(i,i) should also be checked (only for get_r, not for get_mu).

Comment (WvanWoerden) [reply tests/test_gso.py:154]: Other than that this PR can be merged.

Comment (jetSett) [reply tests/test_gso.py:156]: Fixed in b0d4ca43b8e66e7fcc3f63519c85b1b261f4defc

**Pull request merged**